Source/Scene/PointCloudPostProcessor.js

```diff
@@ -0,0 +1,1178 @@
+            this.neighborhoodHalfWidth = tileset.pointCloudPostProcessorOptions.neighborhoodHalfWidth;
+            this.numRegionGrowingPasses = tileset.pointCloudPostProcessorOptions.numRegionGrowingPasses;
+            this.densityHalfWidth = tileset.pointCloudPostProcessorOptions.densityHalfWidth;
+            this.neighborhoodVectorSize = tileset.pointCloudPostProcessorOptions.neighborhoodVectorSize;
+            this.densityViewEnabled = tileset.pointCloudPostProcessorOptions.densityViewEnabled;
+            this.depthViewEnabled = tileset.pointCloudPostProcessorOptions.depthViewEnabled;
+            this.stencilViewEnabled = tileset.pointCloudPostProcessorOptions.stencilViewEnabled;
+            this.distanceConstraint = tileset.pointCloudPostProcessorOptions.distanceConstraint;
+            this.maxAbsRatio = tileset.pointCloudPostProcessorOptions.maxAbsRatio;
+            this.pointAttenuationMultiplier = tileset.pointCloudPostProcessorOptions.pointAttenuationMultiplier;
+            this.useTriangle = tileset.pointCloudPostProcessorOptions.useTriangle;
+            this.enableAO = tileset.pointCloudPostProcessorOptions.enableAO;
+            this.aoViewEnabled = tileset.pointCloudPostProcessorOptions.aoViewEnabled;
+            this.sigmoidDomainOffset = tileset.pointCloudPostProcessorOptions.sigmoidDomainOffset;
+            this.sigmoidSharpness = tileset.pointCloudPostProcessorOptions.sigmoidSharpness;
+            this.dropoutFactor = tileset.pointCloudPostProcessorOptions.dropoutFactor;
+            this.delay = tileset.pointCloudPostProcessorOptions.delay;
+            dirty = true;
+        }
+
+        dirty |= createResources(this, frameState.context, dirty);
+
+        // Render point cloud commands into an offscreen FBO.
+        var i;
+        var commandList = frameState.commandList;
+        var commandEnd = commandList.length;
+
+        var attenuationMultiplier = this.pointAttenuationMultiplier;
+        var attenuationUniformFunction = function() {
+            return attenuationMultiplier;
+        };
+
+        for (i = commandStart; i < commandEnd; ++i) {
+            var command = commandList[i];
+            if (command.primitiveType !== PrimitiveType.POINTS) {
+                continue;
+            }
+
+            var derivedCommand = command.derivedCommands.pointCloudProcessor;
+            if (!defined(derivedCommand) || command.dirty || dirty) {
+                derivedCommand = DrawCommand.shallowClone(command);
+                command.derivedCommands.pointCloudProcessor = derivedCommand;
+
+                derivedCommand.framebuffer = this._framebuffers.prior;
+                derivedCommand.shaderProgram = getECShaderProgram(frameState.context, command.shaderProgram);
+                derivedCommand.castShadows = false;
+                derivedCommand.receiveShadows = false;
+
+                var derivedCommandRenderState = RenderState.fromCache(derivedCommand.renderState);
```

Comment:
Do a `clone` here instead since the `RenderState.fromCache` call is below. For example, look at https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/Cesium3DTileBatchTable.js#L1329-L1332.